src/core/src/op/fake_convert.cpp

```diff
@@ -39,37 +40,28 @@ const std::string& FakeConvert::get_destination_type() const {
 
 void FakeConvert::validate_and_infer_types() {
     OV_OP_SCOPE(v13_FakeConvert_validate_and_infer_types);
-    validate_type();
+    validate_destination_type();
+    auto out_type = ov::element::Type(element::dynamic);
     for (size_t i = 0; i < get_input_size(); i++) {
-        switch (get_input_element_type(i)) {
-        case element::bf16:
-        case element::f16:
-        case element::f32:
-        case element::dynamic:
-            break;
-        default:
-            OPENVINO_THROW("The element type of the input tensor on index ",
-                           i,
-                           " must be a bf16, f16, f32 or dynamic (got ",
-                           get_input_element_type(i),
-                           ").");
-        }
+        OPENVINO_ASSERT(element::Type::merge(out_type, out_type, get_input_element_type(i)),
+                        "Mixed input types are not supported.");
     }
-    if (inputs().size() == 3) {
-        OPENVINO_ASSERT(get_input_partial_shape(1).compatible(get_input_partial_shape(2)),
-                        "FakeConvert scale shape: ",
-                        get_input_partial_shape(1),
-                        " is not compatible with shift shape: ",
-                        get_input_partial_shape(2));
+    switch (out_type) {
```

Comment:
Make helper for it and use in macro as function.

Reply:
Not sure if separating this switch into function would have any sense. Looking on has_evaluate, it couldn't be reused there since dynamic dimension is not supported there. Maybe we could consider it in separate PR to somewhat unify it?

Reply:
Yes can be done later.
Dynamic type is not an issue, the function can check what is common check for both cases (there will be one switch case) and cane be expanded locally or by other function by check dynamic type.